ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPublic.qll

```diff
@@ -559,3 +710,618 @@ abstract deprecated class BarrierGuard extends CfgNodes::ExprCfgNode {
     result.asExpr() = this.getAMaybeGuardedCapturedDef().getARead()
   }
 }
+
+/**
+ * A representation of a run-time module or class.
+ *
+ * This is equivalent to the type `Ast::Module` but provides data-flow specific methods.
+ */
+class ModuleNode instanceof Module {
```

Comment:
Can you explain your thinking behind making this an AST node and not a `DataFlow::Node`? I note that JS's `DataFlow::ClassNode` extends `DataFlow::Node`.